db/schema.rb

```diff
@@ -501,9 +501,13 @@
     t.integer "totchild"
     t.integer "hhtype"
     t.integer "hodate_check"
+    t.integer "extrabor_value_check"
+    t.integer "deposit_and_mortgage_value_check"
+    t.integer "grant_value_check"
```

Comment:
Is this meant to be here?

Reply:
as a more experienced rails developer please let me know if you know why this keeps happening in my branches and what the solution is broadly.
It was my understanding that the schema file is auto-generated and generally I should just let it do what it wants, when I migrate it should add/remove/amend the relevant things, and sometimes it reorders some and that isn't really a problem.
I was taught when first using rails that it shouldn't be necessary and is generally a cad idea to edit the schema file yourself as it is auto generated. I've rerun db:migrate to give it a chance to fix itself and it seems to be happy and the server runs, the tests run and pass etc etc so that made me inclined to just leave it.
Thoughts?

Reply:
If you drop and recreate (db:drop, db:create, db:migrate) the database that should fix itself. I think it might be happening when you've had different migrations on different branches. If you had ran the migration on your db on one branch,  switched branches and ran migration on another branch, that would update the schema file to represent the state of your db, which now would have changes from both branches (I think). The reason it didn't fix itself when migrating is probably because the migration number in your db is the same as the migration number in your schema file, so it probably thinks it's up to date